pkg/stack_mgr/stack_service.go

```diff
@@ -20,8 +20,8 @@ import (
 
 type StackServiceIface interface {
 	NewStackMeta(stackName string) *StackMeta
```

Comment:
I know this would require a large refactor of what you have, but since we already have this interface, I wonder if instead of adding the dryRun bool to these methods, make a separate implementation of `StackServiceIface` that is called something like `StackServiceDryRunner` and when the dry-run flag is set in the command, NewStackService() returns the `StackServiceDryRunner`, else it returns a normal `StackService`. `StackServiceDryRunner` would handle all the implementation details of running these functions in dry-run mode. Some psedocode:

~~~go
func NewStackService(dryRun bool) *StackService {
        if dryRun { 
              return &StackServiceDryRunner{}
        }
	// TODO pass this in instead?
	dirProcessor := util.NewLocalProcessor()

	return &StackService{
		stacks:       nil,
		dirProcessor: dirProcessor,
	}
}
~~~